test/test_page_without_a_file.rb

```diff
@@ -28,6 +28,21 @@ def render_and_write
                        ))
     end
 
+    should "have non-frozen path and relative_path attributes" do
+      {
+        ["foo", "bar.md"]     => "foo/bar.md",
+        [nil, nil]            => "",
+        ["", ""]              => "",
+        ["/lorem/", "/ipsum"] => "lorem/ipsum",
+        %w(lorem ipsum)       => "lorem/ipsum",
+      }.each do |(dir, name), result|
+        page = PageWithoutAFile.new(@site, @site.source, dir, name)
+        assert_equal result, page.path
+        assert_equal result, page.relative_path
+        refute page.relative_path.frozen?
```

Comment:
Is the relative path modified? Why is it important that it's unfrozen?

Reply:
For backwards-compatibility. It is an unfrozen string in v4.1.
`Jekyll::Pathfinder.join` returns a frozen cached string. So this test confirms that `page.relative_path` hasn't changed behavior in v4.2..